system/GlobalCompletionEvent.hpp

```diff
@@ -286,8 +327,23 @@ class GlobalCompletionEvent : public CompletionEvent {
       }
       DVLOG(3) << "fell thru conservative check";
     }
-    CHECK(!event_in_progress) << " gce(" << this << ", count:" << count << ")";
-    CHECK_EQ(count, 0);
+  }
+
+  /// Set up GCE to re-enroll this many cores/tasks on master core at completion.
+  /// This should be called on a single core after enrolling as many tasks as you want to reenroll on the master core (0).
```

Comment:
Well and the weirdest thing about that is not knowing where to send the completions to... Somehow you have to keep track of where all the auto-re-enrolls are and remember to send completions to the right places every time.